package-lock.json

```diff
@@ -9,6 +9,7 @@
 			"version": "0.0.1",
 			"dependencies": {
 				"@preact/signals": "1.1.1",
+				"@types/node": "^18.14.2",
```

Comment:
is this just to get `NodeJS.EventEmitter` type? Could we get it without the dependency?

Reply:
See https://github.com/ethers-io/ethers.js/pull/3539/files for an example of how to get types that differ *slightly* between NodeJS and Browser but are used in the same places.  This example is for NodeJS.Timer, but the same concept may work for different types between NodeJS event emitter and browser event emitter.

Reply:
```ts
interface Eip1193Provider {
	request(request: { method: string; params?: Array<any> | Record<string, any> }): Promise<any>
	on(eventName: string | symbol, listener: (...args: any[]) => void): this
	removeListener(eventName: string | symbol, listener: (...args: any[]) => void): this
}

declare global {
	interface Window {
		ethereum?: Eip1193Provider
	}
}
```
Grabbed the on and removeListener definitions, these are the only ones required by EIP1193